Sources_v3/Workers/ChannelUpdater_Tests.swift

```diff
@@ -71,6 +71,20 @@ class ChannelUpdater_Tests: StressTestCase {
         AssertAsync.willBeEqual(completionCalledError as? TestError, error)
     }
 
+    func test_updateChannelQuery_completionForCreatedChannelCalled() {
+        // Simulate `update(channelQuery:)` call
+        let query = ChannelQuery(channelPayload: .unique)
+        var cid: ChannelId?
+        queryUpdater.update(channelQuery: query, channelCreatedCallback: { cid = $0 }, completion: nil)
+
+        // Simulate API response with channel data
+        let payload = dummyPayload(with: query.cid)
+        apiClient.test_simulateResponse(.success(payload))
+
+        // Assert the `channelCreatedCallback` is called
+        AssertAsync.willBeEqual(query.cid, cid)
+    }
```

Comment:
I think we should add more asserts to this test. We need to test the payload is not saved before the `channelCreatedCallback` returns. It will be a little bit tricky, but it's extremely important since that's the whole point of this hook.